pgmpy/factors/Factor.py

```diff
@@ -159,6 +161,7 @@ def get_cardinality(self, variables):
 
         return {var: self.cardinality[self.variables.index(var)] for var in variables}
 
+    @stateNameDecorator(argument=None, return_val=True)
```

Comment:
@yashu-seth It's also possible that in some case we return more than one values and out of them one of them is a state. You will need to handle that case.